src/plugins/isup.lua

```diff
@@ -0,0 +1,36 @@
+url = require("socket.url")
+https = require("ssl.https")
+
+
+local plugin = {}
+
+local messages = { "The webserver hasn't sent a final response, status %s",
+                   "The website is up status %s",
+                   "The website is up with a %s redirect",
+                   "Client error, webserver returned status %s",
+                   "Server returning error %s" }
+
+plugin.help = 'Usage: isup <website>'
+
+plugin.main = function(args)
+    local website = args.message
+    if args.modules.config.debug then
+        print(website)
+    end
+    if not website then
+        args.modules.irc.privmsg(args.target, 'Give me a website or hostname to check')
+    end
+    local address = website:find('^https?://') and website or ('http://%s'):format(website)
```

Comment:
I like this simplification, but now seeing that `address` is only used on the next line, can we just inline it?

something like this:

```lua
local response, httpcode = https.request(('%s%s'):format(website:find('^https?://') and '' or 'http://', website))
```
Does that feel too compacted?

Reply:
This seems a bit harder to read, but maybe if its broken into more than one line?
```
local response, httpcode = https.request(('%s%s')
                           :format(website:find('^https?://') and '' or 'http://', website))
```

I'm not totally sure how to do multiple lines with lua

Reply:
eh, let's not worry about it for now.